tests/test_build_linkcheck.py

```diff
@@ -95,6 +101,15 @@ def test_defaults(app):
     assert rowsby["http://localhost:7777#does-not-exist"]["info"] == "Anchor 'does-not-exist' not found"
     # images should fail
     assert "Not Found for url: http://localhost:7777/image.png" in rowsby["http://localhost:7777/image.png"]["info"]
+    # anchor should be found
+    assert rowsby['http://localhost:7777/anchor.html#found'] == {
+        'filename': 'links.rst',
+        'lineno': 14,
+        'status': 'working',
+        'code': 0,
+        'uri': 'http://localhost:7777/anchor.html#found',
+        'info': '',
+    }
```

Comment:
Thanks for the review, @francoisfreitag.

> Isn’t that case covered by
> 
> https://github.com/sphinx-doc/sphinx/blob/30f347226fa4ccf49b3f7ef860fd962af5bdf4f0/tests/roots/test-linkcheck/links.txt#L5

Although the linkchecker should look for the anchor element when checking that URL, I don't think that there's test coverage on it.

There are a couple of [`test_defaults*` test cases](https://github.com/sphinx-doc/sphinx/blob/30f347226fa4ccf49b3f7ef860fd962af5bdf4f0/tests/test_build_linkcheck.py#L27-L104) -- but they don't (didn't?) check any positive-found anchor cases.

> Side note, if we want to speedup linkcheck tests, we may replace links going to servers on the internet with going to the local HTTP server, its latency should be lower and very likely faster.

Yep, a good idea.  Commit 97f07ca83c70bba55fe1c1cb25993e5240e4187d should handle that.  It does look like there are still [two (broken link) public URLs](https://github.com/sphinx-doc/sphinx/tree/97f07ca83c70bba55fe1c1cb25993e5240e4187d/tests/roots/test-linkcheck-documents_exclude) in the test data -- but those [should be `ignored`](https://github.com/sphinx-doc/sphinx/blob/97f07ca83c70bba55fe1c1cb25993e5240e4187d/tests/test_build_linkcheck.py#L696-L713) according to the relevant linkchecker test.

Reply:
Well spotted :+1: 

Since the Internet URLs have been replaced by `localhost` in the development branch recently, it feels like it may be easiest to add equivalent anchor-reference coverage using a `localhost` URL.  What do you think?

Reply:
Works for me, thanks!

Reply:
I believe this would be a great addition. Don’t feel like you should take it all by yourself, adding an issue is a good way to keep track of ideas for improvement, and opens the door for others to tackle the issue.

Reply:
Filed as sphinx-contrib/sphinx-lint#62.